rpc/client.go

```diff
@@ -128,9 +130,29 @@ func waitForRequest[T serde.RequestPayload, U serde.ResponsePayload](rc *RpcClie
 	return res.Payload
 }
 
-func (rc *RpcClient) WaitForObjectiveCompletion(expectedObjectiveId protocols.ObjectiveId) {
+func (rc *RpcClient) WaitForObjectiveCompletion(expectedObjectiveId ...protocols.ObjectiveId) {
+	completed := make(map[protocols.ObjectiveId]bool)
+
 	for receivedObjectiveId := range rc.CompletedObjectives() {
-		if expectedObjectiveId == receivedObjectiveId {
+		isObjectiveExpected := false
+		for _, expectedObjectiveId := range expectedObjectiveId {
+			if receivedObjectiveId == expectedObjectiveId {
+				isObjectiveExpected = true
+			}
+		}
+		if !isObjectiveExpected {
+			err := fmt.Errorf("received unexpected objective completion notification for objective %v", receivedObjectiveId)
+			panic(err)
```

Comment:
Is it good to panic here?

Reply:
Good call. `WaitForObjectiveCompletion` now returns an error in [a4d5507](https://github.com/statechannels/go-nitro/pull/1125/commits/a4d550714ca6ceed03347a4dfdaac84d17c92bd0)